doc/ipythonconsole.rst

```diff
@@ -17,79 +12,81 @@ IPython's many features.
+* The :ref:`User Module Reloader <umr-section>`, which can automatically re-import modified packages and files
+* Inline display of ``Matplotlib`` graphics, if the ``Inline`` backend is selected under :menuselection:`Preferences --> IPython console --> Graphics --> Graphics backend`
 
+For information on the features, commands and capabilities built into ``IPython`` itself, see the `IPython documentation`_.
 
-Reloading modules: the User Module Reloader (UMR)
--------------------------------------------------
+.. _IPython documentation: https://ipython.readthedocs.io/en/stable/overview.html
 
-When working with Python scripts interactively, one must keep in mind that
-Python import a module from its source code (on disk) only when parsing the
-first corresponding import statement. During this first import, the byte code
-is generated (.pyc file) if necessary and the imported module code object is
-cached in `sys.modules`. Then, when re-importing the same module, this cached
-code object will be directly used even if the source code file (.py[w] file)
-has changed meanwhile.
 
-This behavior is sometimes unexpected when working with the Python interpreter
-in interactive mode, because one must either always restart the interpreter
-or remove manually the .pyc files to be sure that changes made in imported
-modules were taken into account.
+.. _umr-section:
 
-The User Module Reloader (UMR) is a Spyder console's exclusive feature that
-forces the Python interpreter to reload modules completely when executing
-a Python script.
+===================================
+Using UMR to reload changed modules
+===================================
 
-For example, when UMR is turned on, one may test complex applications
-within the same Python interpreter without having to restart it every time
-(restart time may be relatively long when testing GUI-based applications).
+When working with scripts and modules in an interactive session, Python only loads a module from its source file once, the first time it is ``import``ed.
+During this first ``import``, the bytecode (``.pyc`` file) is generated if necessary and the imported module object is cached in ``sys.modules``.
+If you subsequently re-import the module anytime in the same session, this cached code object will be used even if its source code (``.py{w}`` file) has changed in the meantime.
+While efficient for final production code, this behavior is often undesired when working with the Python interpreter interactively, such as when analyzing data or testing your own module.
+In effect, you're left with no way to update or modify any already-imported modules, aside from manually removing the relevant ``.pyc`` files, or completely restarting the console entirely.
```

Comment:
Let's remove `completely` here.

Reply:
Two adverbs in one sentence seems too much to me :-)

Reply:
Yup :) Good catch; fixed.